webui/react/src/components/LogViewerTimestamp.tsx

```diff
@@ -478,44 +451,41 @@ const LogViewerTimestamp: React.FC<Props> = ({
     };
   }, [
     addLogs,
-    clearLogs,
     direction,
     fetchToLogConverter,
     filter,
-    onFetchLogAfter,
     onFetchLogTail,
   ]);
 
   /*
-   * Load older/newer log entries when the user scroll to the top/bottom.
+   * Load old Log entries (api no-follow) when container scroll is at the top or bottom.
    */
   useEffect(() => {
     if (isLastReached) return;
 
     const canceler = new AbortController();
+    const logTimes = logs.map(log => log.time).sort();
     let fetchArgs = null;
     let isPrepend = false;
 
-    if (direction === DIRECTIONS.TAILING) {
-      if (!logTimestampRange.min) return;
-      if (!isScrollOnTop) return;
-      if (isPrevScrollOnTop) return;
-
+    if (direction === DIRECTIONS.TAILING && isScrollOn.top) {
+      const firstLogTime = logTimes[0];
       fetchArgs = onFetchLogBefore({
         ...filter,
-        timestampBefore: dayjs(logTimestampRange.min),
+        timestampBefore: (
+          firstLogTime ? dayjs(firstLogTime).add(1, 'second') : filter.timestampBefore
+        ),
```

Comment:
**suggestion:** I see, so you are adding a second to the before to ensure an overlap in case there are entries right in that sliver where it can potentially get dropped and our dedupe will process to filter out the duplicates. How about adding just `1 millisecond` since we now support that resolution?

Reply:
now that we've upped the `TAIL_SIZE` to 500, the odds of having more than that number of events within 1 second diminishes but I think changing this to `1 millisecond` overlap will ensure that we don't get stuck in requesting the same range of data.

Reply:
1. Reduced TAIL_SIZE to 200, consider in the other LogViewer it's 1000.
2. Consider a case where you have 2 logs at the same milliseconds and limit/offset cut the return exactly so that you get only 1 of those, you want the api to start from the last millisecond (included) so that you get the other one too. Since `timestampBefore` is inclusive, we need that only on `timestampAfter`.